UnleashClient/events.py

```diff
@@ -11,17 +12,51 @@ class UnleashEventType(Enum):
 
     FEATURE_FLAG = "feature_flag"
     VARIANT = "variant"
+    FETCHED = "fetched"
+    READY = "ready"
 
 
 @dataclass
-class UnleashEvent:
+class BaseEvent:
     """
-    Dataclass capturing information from an Unleash feature flag or variant check.
+    Base event type for all events in the Unleash client.
     """
 
     event_type: UnleashEventType
     event_id: UUID
+
+
+@dataclass
+class UnleashEvent(BaseEvent):
+    """
+    Dataclass capturing information from an Unleash feature flag or variant check.
+    """
+
     context: dict
     enabled: bool
     feature_name: str
     variant: Optional[str] = ""
+
+
+@dataclass
+class UnleashReadyEvent(BaseEvent):
+    """
+    Event indicating that the Unleash client is ready.
+    """
+
+    pass
+
+
+@dataclass
+class UnleashFetchedEvent(BaseEvent):
+    """
+    Event indicating that the Unleash client has fetched feature flags.
+    """
+
+    raw_features: str
+
+    @property
+    def features(self) -> dict:
+        if not hasattr(self, "_parsed_payload"):
+            self._parsed_payload = loads(self.raw_features)["features"]

```

Comment:
Annoyingly this now becomes part of the public API.

Kinda struggled a little with choosing between only returning the features vs returning the whole response. Leaning ever so slightly in the direction of only returning the features since its less confusing for end users.

That being said, every time we've done that in the past (fetch features, bootstrap, load from backup), it's been a mistake that needs to be patched later. Open to being told this is similar but I can't see how right now